_includes/manuals/2.1/1.2_release_notes.md

```diff
@@ -389,6 +389,9 @@ By default it is deployed using [systemd socket activation](http://0pointer.de/b
 
 ### Deprecations
 
+* Puppetrun feature is being removed from Foreman core since version 2.2 ([more info](https://community.theforeman.org/t/dropping-run-puppet-button-with-a-replacement-using-remote-execution-plugin/12781/25)). However, the function of Puppetrun can be subsided by Remote Execution plugin. There is two options:
+    * There is `Puppet run once - SSH default` job template that takes `puppet_options` as parameter before it's invoked
+    * Another option is for `mcollective`/`choria` users. The best way is write own job template that leverages `load_hosts` macro for choosing hosts and then this selection is used as a parameter for `mco` program, for ie. `mco ... --runs <%= load_hosts(input('search')).join(',') %>`
```

Comment:
Is this already working? @ares  mentioned in his last post that we need to make `load_hosts` macro available for remote execution templates, I'm not sure it has been done already. It would be good if we have an example template in REX already rather than giving users an example here that hasn't been tested to work.
It should also be mentioned that this option requires triggering the job on a host that has these tools installed and has access to, not on the

Reply:
I defer to verify this to @domitea, however I think it should already be available, loader macros are included in [base scope](https://github.com/theforeman/foreman/blob/develop/lib/foreman/renderer/scope/base.rb#L7) since it's usefull for all template types.

I think example is fine if we test it can generate the list of hosts. We don't know how exactly people want to use this (mco vs choria). If we see the interest, we can build the Foreman official template once we hear, how users actually do it.

The more real life example though would be

```erb
 mco rpc puppetd runonce -I <%= load_hosts(input('hosts')).join(' ') %>
```

Reply:
The `load_hosts` macro is working in job templates because as @ares wrote, it's in base scope. I haven't tested real-life example due to my lack of puppet knowledge.